api/src/main/java/jakarta/servlet/http/HttpServlet.java

```diff
@@ -143,7 +145,7 @@ public HttpServlet() {
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         String protocol = req.getProtocol();
         String msg = lStrings.getString("http.method_get_not_supported");
-        if (protocol.endsWith("1.1")) {
+        if (!protocol.endsWith(PROTOCOL_VERSION_1_0)) {
```

Comment:
I know that HTTP 0.9 is really meant to be supported anymore, but there are probably still some usages out there.
Perhaps a switch approach is better:
```suggestion
        switch(protocol) {
          case "HTTP/0.9":
          case "HTTP/1.0":
            resp.sendError(HttpServletResponse.SC_BAD_REQUEST, msg);
            break;
          default:
            resp.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED, msg);
            break;
        }
```

Reply:
yes, good point, I've modified changes accordingly